### ISSUE TITLE
[Bug]: ERR_REQUIRE_ESM with base setup

### ISSUE THREAD

**Issue by apstone** [serwist/serwist#22]
### Provide environment information

```
    "@serwist/next": "^8.2.0",
    "@serwist/precaching": "^8.2.0",
    "@serwist/sw": "^8.2.0",
```

### Which project is this issue for?

@serwist/next

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code help us address them faster

https://codesandbox.io/p/devbox/eloquent-kate-26hzzz?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clq5cq35i00093b6r2phk3uh3%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clq5cq35h00023b6rt4lzp68k%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clq5cq35h00063b6rwfki12rr%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clq5cq35h00083b6ryjyal06n%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clq5cq35h00023b6rt4lzp68k%2522%253A%257B%2522id%2522%253A%2522clq5cq35h00023b6rt4lzp68k%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clq5cq35h00083b6ryjyal06n%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clq5cq35h00073b6rdm9bxsyu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clq5cq35h00083b6ryjyal06n%2522%252C%2522activeTabId%2522%253A%2522clq5cq35h00073b6rdm9bxsyu%2522%257D%252C%2522clq5cq35h00063b6rwfki12rr%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clq5cq35h00033b6rmkg06pgy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clq5cq35h00043b6rv4z80cli%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%252C%257B%2522id%2522%253A%2522clq5cq35h00053b6rc94ys9ea%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%257D%255D%252C%2522id%2522%253A%2522clq5cq35h00063b6rwfki12rr%2522%252C%2522activeTabId%2522%253A%2522clq5cq35h00033b6rmkg06pgy%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### To reproduce

`npm run build` following basic instructions

### Describe the bug

Doesn't build -

```
$ npm run build

> build
> next build

 ⨯ Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error

> Build error occurred
Error [ERR_REQUIRE_ESM]: require() of ES Module C:\Users\apsto\dev\website\node_modules\@serwist\webpack-plugin\dist\relative-to-output-path-5MVPPhSi.js from C:\Users\apsto\dev\website\node_modules\@serwist\webpack-plugin\dist\index.cjs not supported.
relative-to-output-path-5MVPPhSi.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead either rename relative-to-output-path-5MVPPhSi.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in C:\Users\apsto\dev\website\node_modules\@serwist\webpack-plugin\package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at mod.require (C:\Users\apsto\dev\website\node_modules\next\dist\server\require-hook.js:64:28)
    at Object.<anonymous> (C:\Users\apsto\dev\website\node_modules\@serwist\webpack-plugin\dist\index.cjs:8:28) {
  code: 'ERR_REQUIRE_ESM'
  ```

### Expected behavior

Build succeeds

### Additional information (if relevant)

8.04 Seems to build fine

**Comment by apstone** [comment]
Happy to help if you will point me in the right direction!

**Comment by DuCanhGH** [comment]
Bloody hell.
Supporting both ESM and CJS is just so much hassle I'm not going to lie.
This was due to a change in our rollup.config.js:
Old:
```
[
  {
    input: "src/index.ts",
    output: [
      {
        file: "dist/index.old.cjs",
        format: "cjs",
        exports: "named",
      },
      {
        file: "dist/index.js",
        format: "esm",
      },
    ],
  },
  {
    input: "src/index.internal.ts",
    output: [
      {
        file: "dist/index.internal.old.cjs",
        format: "cjs",
        exports: "named",
      },
      {
        file: "dist/index.internal.js",
        format: "esm",
      },
    ],
  },
];
```
New:
```
{
  input: {
    index: "src/index.ts",
    "index.internal": "src/index.internal.ts",
  },
  output: [
    {
      dir: "dist",
      entryFileNames: "[name].cjs",
      format: "cjs",
      exports: "named",
    },
    {
      dir: "dist",
      entryFileNames: "[name].js",
      format: "esm",
    },
  ],
}
```
Turns out I also need `chunkFileNames`, otherwise Rollup would always use ".js" for all the chunks. What in the world.

**Comment by DuCanhGH** [comment]
@apstone should be fixed in 8.3.0. Give it a try!

**Comment by apstone** [comment]
> @apstone should be fixed in 8.3.0. Give it a try!

Looks to be working again for me. Thanks again @DuCanhGH
